tests/assets/crs/bayesian-optimization.yaml

```diff
@@ -0,0 +1,70 @@
+---
+apiVersion: kubeflow.org/v1beta1
+kind: Experiment
+metadata:
+  name: bayesian-optimization-annot
+spec:
+  objective:
+    type: maximize
+    goal: 0.99
+    objectiveMetricName: Validation-accuracy
+    additionalMetricNames:
+      - Train-accuracy
+  algorithm:
+    algorithmName: bayesianoptimization
+    algorithmSettings:
+      - name: "random_state"
+        value: "10"
+  parallelTrialCount: 1
+  maxTrialCount: 1
+  maxFailedTrialCount: 1
+  parameters:
+    - name: lr
+      parameterType: double
+      feasibleSpace:
+        min: "0.01"
+        max: "0.03"
+    - name: num-layers
+      parameterType: int
+      feasibleSpace:
+        min: "2"
+        max: "5"
+    - name: optimizer
+      parameterType: categorical
+      feasibleSpace:
+        list:
+          - sgd
+          - adam
+          - ftrl
+  trialTemplate:
+    primaryContainerName: training-container
+    trialParameters:
+      - name: learningRate
+        description: Learning rate for the training model
+        reference: lr
+      - name: numberLayers
+        description: Number of training model layers
+        reference: num-layers
+      - name: optimizer
+        description: Training model optimizer (sdg, adam or ftrl)
+        reference: optimizer
+    trialSpec:
+      apiVersion: batch/v1
+      kind: Job
+      spec:
+        template:
+          spec:
+            containers:
+              - name: training-container
+                image: docker.io/kubeflowkatib/mxnet-mnist:v0.15.0
+                command:
+                  - "python3"
+                  - "/opt/mxnet-mnist/mnist.py"
+                  - "--batch-size=64"
+                  - "--lr=${trialParameters.learningRate}"
+                  - "--num-layers=${trialParameters.numberLayers}"
+                  - "--optimizer=${trialParameters.optimizer}"
+            restartPolicy: Never
+          metadata:
+            annotations:
+              sidecar.istio.io/inject: "false"
```

Comment:
it's actually not needed at all in the kubeflow namespace, let me remove it.

Reply:
Oh, wait I didn't catch that we're running these in the `kubeflow` namespace. Why are we doing that? Users would never run workloads in that namespace!

Reply:
would it make sense to create a user namespace and apply experiments to it? then we'll need to set sidecar injection to false in all the crs.

Reply:
Yes, I believe it would be better if we created a user Profile (the namespace will be created by the corresponding controller) and ran the Experiments there. And yes, we would then set sidecar injection to false everywhere!

Reply:
This is what we do for the UATs BTW: https://github.com/canonical/charmed-kubeflow-uats/blob/main/driver/test_kubeflow_workloads.py

Reply:
alright

Reply:
Let's not deploy profiles operator in these tests as it would be a stretch, we can simply create the namespace manually and add the label.

Reply:
after revisiting this discussion, creating a profile and running experiments in that profile's namespace is a depiction of user experience, so changes were made to use profiles-operator rather than manually create the namespace.